Jellyfin.Plugin.AniDB/Providers/equals_check.cs

```diff
@@ -14,534 +12,223 @@ namespace Jellyfin.Plugin.AniDB.Providers
-                            pre_aid.Add(s2);
-                        }
+                        results.Add(s);
                     }
                     x++;
                 }
-                foreach (string _aid in pre_aid)
-                {
-                    XElement doc = await Task.Run(async () => XElement.Parse("<?xml version=\"1.0\" encoding=\"UTF - 8\"?>" + "<animetitles>" + await One_line_regex(await Task.Run(() => new Regex("<anime aid=\"" + _aid + "\">" + @"(?s)(.*?)<\/anime>"), cancellationToken), xml, cancellationToken, 0) + "</animetitles>"), cancellationToken);
-                    var a_ = from page in doc.Elements("anime")
-                             where _aid == page.Attribute("aid").Value
-                             select page;
-                    if (await Simple_compare( a_.Elements("title"), b, cancellationToken) && await Simple_compare(a_.Elements("title"), a, cancellationToken))
-                    {
-                        return true;
-                    }
-                }
-                return false;
             }
             catch (Exception)
             {
-                if (retry)
-                {
-                    return false;
-                }
-                else
+                if (x_ == 0)
                 {
                     await Task.Run(() => AniDbTitleDownloader.Load_static(cancellationToken), cancellationToken);
-                    return await Fast_xml_search(a, b, cancellationToken, false, true);
+                    return await XmlSearch(name, cancellationToken, 1);
                 }
             }
+
+            return results;
         }
 
         /// <summary>
-        /// Return the AniDB ID if a and b match
+        /// Finds an AniDB ID for name
         /// </summary>
-        public async static Task<string> Fast_xml_search(string a, string b, CancellationToken cancellationToken, bool return_AniDBid, int x_ = 0)
+        public async static Task<string> XmlFindId(string name, CancellationToken cancellationToken, int x_ = 0)
         {
-            //Get AID aid=\"([s\S].*)\">
-            try
+            var results = await XmlSearch(name, cancellationToken);
+
+            if (results.Count == 1)
             {
-                List<string> pre_aid = new List<string>();
-                string xml = File.ReadAllText(Get_anidb_xml_file());
-                int x = 0;
-                string s1 = "-";
-                string s2 = "-";
-                while (!string.IsNullOrEmpty(s1) && !string.IsNullOrEmpty(s2))
+                return results[0];
+            }
+
+            string xml = File.ReadAllText(GetAnidbXml());
+            int lowestDistance = Plugin.Instance.Configuration.TitleSimilarityThreshold;
+            string currentId = "";
+            foreach (string id in results)
+            {
+                string nameXmlFromId = OneLineRegex(new Regex(@"<anime aid=""" + id + @"""((?s).*?)<\/anime>", RegexOptions.Compiled), xml);
```

Comment:
This regex should be initialized once

Reply:
How would that work? It needs the id.

Reply:
Probably shouldn't be a regex then :)
It's fine for now since it wasn't an addition